Marlin/src/module/temperature.cpp

```diff
@@ -3216,64 +3229,66 @@ void Temperature::tick() {
       }
       break;
 
+    // TODO: Make an array of those, by merging them to
+    // a single state instead of so many different state
     #if HAS_TEMP_ADC_0
```

Comment:
@thinkyhead : Is the order of reading the sensors important here ? 
I can probably rework these 60 lines in less than 4 here by replacing with a single state and a sub-state counter on the heater's array index. 
But this means following the heater's declaration order, that is, all hotends then temp probe, then bed, then chamber, then cooler.

Reply:
Not the order itself, but you need some time for the adc to settle. So every time you switch adc (HAL_START_ADC) you need to wait before ACCUMULATE_ADC. Therefore one state for each.

Reply:
Ok, that's the confirmation I was expecting. Thanks!